devito/dse/rewriters.py

```diff
@@ -406,6 +408,47 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        A pass to perform only the basic skewing in order to have valid data
+        dependences.
+        """
+        skew_factor = configuration['skew_factor']
+        skew_dim, mapper, int_mapper = None, {}, []
+        total_int = {}
+        intervals, sub_iterators, directions = cluster.ispace.args
+        cnt = 0
+        for dim in cluster.ispace.dimensions:
+            if isinstance(dim, SpaceDimension):
+                mapper[dim] = dim + skew_factor*skew_dim
+                int_mapper.append(IntervalGroup([Interval(dim, -skew_factor*skew_dim,
+                                                -skew_factor*skew_dim)]))
+                cnt = cnt + 1
+            elif dim.is_Time:
+                if isinstance(dim, TimeDimension):
+                    int_mapper.append(IntervalGroup([Interval(dim, 0, 0)]))
+                    cnt = cnt + 1
+                    skew_dim = dim
+                elif isinstance(dim.parent, TimeDimension):
+                    int_mapper.append(IntervalGroup([Interval(dim, 0, 0)]))
+                    cnt = cnt + 1
+        if skew_dim is None:
+            return cluster
+
+        if cnt == 5:
```

Comment:
I do not like this as I have it here. How can I parametrize somehow the int_mapper[] to pass them like all together
without having 4-5 arguments in a row?

Reply:
fixed